local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -214,14 +216,20 @@ void WaypointGenerator::nextSmoothYaw(float dt) {
   // and we dont want to introduce yet another parameter
 
   const float desired_setpoint_yaw_rad =
-      nextYaw(position_, output_.goto_position);
+      (position_ - output_.goto_position).normXY() > 0.1f
+          ? nextYaw(position_, output_.goto_position)
+          : curr_yaw_rad_;
 
-  if (smoothing_speed_xy_ > 0) {
+  // If smoothing is disabled, set yaw to face goal directly
+  if (smoothing_speed_xy_ <= 0.01f) {
+    setpoint_yaw_rad_ = desired_setpoint_yaw_rad;
+    return;
```

Comment:
I don't think you need the return here, since everything is inside of the `else` afterwards

Reply:
Good point, my idea behind this refactor was reducing the nested if() clauses with an early exit. Similar to SmoothWaypoint(). I'll remove the else